sonar-plugin-api/src/main/java/org/sonar/api/batch/sensor/coverage/internal/DefaultCoverage.java

```diff
@@ -128,6 +147,7 @@ public int coveredConditions() {
   public void doSave() {
     Preconditions.checkNotNull(inputFile, "Call onFile() first");
     Preconditions.checkNotNull(type, "Call ofType() first");
+    validateLines();
```

Comment:
Nothing prevents the plugins to call onFile just before saving, so we might not have the number of lines in the file.